File: core-module/src/main/java/org/artembogomolova/demo/webapp/dao/service/RoleRepoService.java
package org.artembogomolova.demo.webapp.dao.service;

import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;
import lombok.RequiredArgsConstructor;
import org.artembogomolova.demo.webapp.dao.repo.IUserRoleRepository;
import org.artembogomolova.demo.webapp.domain.auth.PredefinedUserRole;
import org.artembogomolova.demo.webapp.domain.auth.Role;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class RoleRepoService {

  private final IUserRoleRepository userRoleRepository;

  public void fillAuthoritiesAndRoles() {
    List<Role> roleList = Arrays.stream(PredefinedUserRole.values()).map(Role::new).collect(Collectors.toList());
    userRoleRepository.saveAll(roleList);
  }
}

File: app/src/main/kotlin/org/artembogomolova/demo/webapp/main/dao/service/RoleRepoService.kt
package org.artembogomolova.demo.webapp.main.dao.service

import java.util.Arrays
import java.util.stream.Collectors
import org.artembogomolova.demo.webapp.main.dao.repo.IRoleRepository
import org.artembogomolova.demo.webapp.main.domain.auth.PredefinedUserRole
import org.artembogomolova.demo.webapp.main.domain.auth.Role
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component
class RoleRepoService @Autowired constructor(
    private val userRoleRepository: IRoleRepository
) {

    fun fillAuthoritiesAndRoles() {
        val roleList: List<Role> = Arrays.stream(PredefinedUserRole.values()).map { predefinedUserRole: PredefinedUserRole -> Role.from(predefinedUserRole) }
            .collect(Collectors.toList())
        userRoleRepository.saveAll(roleList)
    }
}